plugins/out_stackdriver/stackdriver.c

```diff
@@ -447,7 +449,37 @@ static int get_severity_level(severity_t * s, const msgpack_object * o,
     return -1;
 }
 
-static int pack_json_payload(int operation_extracted, int operation_extra_size, 
+static void validate_insertId(msgpack_object * subobj, const msgpack_object * o, insertId_status *status)
+{
+    int i = 0;
+    msgpack_object_kv * p = NULL;
+    *status = INSERTID_NOT_EXISTED;
+
+    if (o == NULL || subobj == NULL) {
+        return;
+    }
+
+    for (i = 0; i < o->via.map.size; i++) {
+        p = &o->via.map.ptr[i];
+        if (p->key.type != MSGPACK_OBJECT_STR) {
+            continue;
+        }
+        if (strncmp(INSERTID_IN_JSON, p->key.via.str.ptr, p->key.via.str.size) == 0) {
```

Comment:
before strncmp(), validate the length of INSERTID_IN_JSON vs p->key.via.str.size